Ble ctlr without mpsl

sofu1 opened the pull request:
Just for running CI.

NordicBuilder commented:
**Some checks failed. Please fix and resubmit.**

## checkpatch issues

```
No additional types will be considered - file 'scripts/checkpatch/typedefsfile': No such file or directory
-:380: ERROR:TRAILING_WHITESPACE: trailing whitespace
#380: FILE: subsys/bluetooth/controller/hci_driver.c:388:
+#ifdef CONFIG_CLOCK_CONTROL_NRF_K32SRC_500PPM //mpsl configs?        $

-:380: ERROR:C99_COMMENTS: do not use C99 // comments
#380: FILE: subsys/bluetooth/controller/hci_driver.c:388:
+#ifdef CONFIG_CLOCK_CONTROL_NRF_K32SRC_500PPM //mpsl configs?        

-:429: ERROR:C99_COMMENTS: do not use C99 // comments
#429: FILE: subsys/bluetooth/controller/hci_driver.c:421:
+	//mpsl_clock_lf_cfg_t clock_cfg;

-:431: ERROR:C99_COMMENTS: do not use C99 // comments
#431: FILE: subsys/bluetooth/controller/hci_driver.c:423:
+	//clock_cfg.rc_ctiv = BLE_CONTROLLER_RECOMMENDED_RC_CTIV;

-:432: ERROR:C99_COMMENTS: do not use C99 // comments
#432: FILE: subsys/bluetooth/controller/hci_driver.c:424:
+	//clock_cfg.rc_temp_ctiv = BLE_CONTROLLER_RECOMMENDED_RC_TEMP_CTIV;

-:433: ERROR:C99_COMMENTS: do not use C99 // comments
#433: FILE: subsys/bluetooth/controller/hci_driver.c:425:
+	//clock_cfg.accuracy = m_config_clock_accuracy_get();

-:434: ERROR:C99_COMMENTS: do not use C99 // comments
#434: FILE: subsys/bluetooth/controller/hci_driver.c:426:
+	//clock_cfg.source = m_config_clock_source_get();

-:436: ERROR:TRAILING_WHITESPACE: trailing whitespace
#436: FILE: subsys/bluetooth/controller/hci_driver.c:428:
+/* temporary nrf clock stuff: */    $

-:445: ERROR:C99_COMMENTS: do not use C99 // comments
#445: FILE: subsys/bluetooth/controller/hci_driver.c:437:
+	//err = mpsl_init(&clock_cfg, SWI5_IRQn, /* MPSL_LOW_PRIO_IRQn, */

-:446: ERROR:C99_COMMENTS: do not use C99 // comments
#446: FILE: subsys/bluetooth/controller/hci_driver.c:438:
+	//		blectlr_assertion_handler);

-:447: ERROR:C99_COMMENTS: do not use C99 // comments
#447: FILE: subsys/bluetooth/controller/hci_driver.c:439:
+	//if (err) {

-:448: ERROR:C99_COMMENTS: do not use C99 // comments
#448: FILE: subsys/bluetooth/controller/hci_driver.c:440:
+	//	return err;

-:449: ERROR:C99_COMMENTS: do not use C99 // comments
#449: FILE: subsys/bluetooth/controller/hci_driver.c:441:
+	//}

- total: 13 errors, 0 warnings, 398 lines checked

NOTE: For some of the reported defects, checkpatch may be able to
      mechanically convert to the typical style using --fix or --fix-inplace.

NOTE: Whitespace errors detected.
      You may wish to use scripts/cleanpatch or scripts/cleanfile

Your patch has style problems, please review.

NOTE: Ignored message types: AVOID_EXTERNS BRACES CONFIG_EXPERIMENTAL CONST_STRUCT DATE_TIME FILE_PATH_CHANGES MINMAX MULTISTATEMENT_MACRO_USE_DO_WHILE NETWORKING_BLOCK_COMMENT_STYLE PRINTK_WITHOUT_KERN_LEVEL SPLIT_STRING VOLATILE

NOTE: If any of the errors are false positives, please report
      them to the maintainers.
```

## Gitlint issues

Commit d50438b19e:
1: UC6 Body has no content, should at least have 1 line.

Commit cf26962bf5:
3: UC4 Line exceeds max length (80>72): "Enables using ble_controller library that does not contain mpsl, so that mpsl is"



*Tip: The bot edits this comment instead of posting a new one, so you can check the comment's history to see earlier messages.*

sofu1 commented:
Draft PR to make the final PR go fast.
This PR is for having ble_controller and MPSL as two separate libraries, i.e. that ble_controller does not contain MPSL but is linked together with it in NCS.
The clock driver (previously clock_control_nrfxlib.c) now only uses MPSL API, so I renamed it and moved it out of bt_ll_nrfxlib.
@rugeGerritsen @thomasstenersen @joerchan @minkargar @frederik-vestre-nordic @anangl

sofu1 commented on lib/multithreading_lock/multithreading_lock.h:
```diff
@@ -7,7 +7,7 @@
  * @file multithreading_lock.h
  *
- * @brief APIs for ensuring BLE controller threadsafe operation.
+ * @brief APIs for ensuring MPSL threadsafe operation.
```
MPSL and ble_controller

sofu1 replied:
or remove both

sofu1 commented on subsys/CMakeLists.txt:
```diff
@@ -1,11 +1,13 @@
 
 if (CONFIG_BT)
   add_subdirectory(bluetooth)
+elseif(CONFIG_MPSL)
```
should be if

sofu1 commented on subsys/bluetooth/controller/hci_driver.c:
```diff
@@ -308,6 +309,7 @@ static void signal_thread(void *p1, void *p2, void *p3)
 	while (true) {
 		k_sem_take(&sem_signal, K_FOREVER);
+		/* Call mpsl_low_priority_process(); when DRGN-12631 is done */
 		ble_controller_low_prio_tasks_process();
```
this will be removed

sofu1 commented on subsys/bluetooth/controller/hci_driver.c:
```diff
@@ -368,53 +370,79 @@ u8_t bt_read_static_addr(bt_addr_le_t *addr)
 }
 
-static int ble_init(struct device *unused)
+static u8_t m_config_clock_source_get(void)
```
will be in mpsl subsys instead

sofu1 commented on subsys/bluetooth/controller/hci_driver.c:
```diff
@@ -565,7 +593,6 @@ static int hci_driver_init(struct device *unused)
 	IRQ_CONNECT(SWI5_IRQn, BLE_CONTROLLER_IRQ_PRIO_LOW,
```
remove

joerchan commented:
What? Is this a duplicate?

sofu1 commented:
> What? Is this a duplicate?

Yes, it was just a draft that I made earlier for the actual PR to go faster
